drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1120 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IFadxl314_init ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdlib.h>
+#include <stdint.h>
+#include <errno.h>
+#include "adxl314.h"
+#include "no_os_print_log.h"
+#include "no_os_util.h"
+#include "no_os_delay.h"
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+static int64_t adxl314_accel_conv(uint16_t raw_accel);
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/*******************************************************************************
+ * @brief Read data from the device.
+ *
+ * @param dev          - The device structure.
+ * @param base_address - Address of the base register.
+ * @param size         - The number of bytes to be read and returned in read_data.
+ * @param read_data    - The read data buffer.
+ *
+ * @return ret         - Result of the reading operation (0 - success, negative
+ *                       value for failure).
+*******************************************************************************/
+int adxl314_read_device_data(struct adxl314_dev *dev, uint8_t base_address,
```

Comment:
I think we should add the `adxl314_reg_read(struct adxl314_dev *dev, uint8_t base_address, uint8_t *read_data)` and `adxl314_reg_write(struct adxl314_dev *dev, uint8_t base_address, uint8_t write_data)` functions. I think we find them more often in other drivers, maybe creating a recognizable pattern, as well as they will be useful for when we add the IIO driver.